src/main/java/com/example/java/maven/cardGamePeterPan/Game.java

```diff
@@ -47,59 +48,86 @@ public void printPlayers() {
 
     public void choseCardsToDiscard(Player player) {
         messagePrinter.printPlayer(player);
+
         messagePrinter.printMessage(ASK_FOR_CARDS_TO_DISCARD);
-        int firstCardToDiscardIndex = (getValidCardIndex(player));
+        int firstCardToDiscardIndex = userInputProvider.getNumberOfChosenCard();
         messagePrinter.printMessage(ASK_FOR_NEXT_CARD_TO_DISCARD);
-        int secondCardToDiscardIndex = (getValidCardIndex(player));
+        int secondCardToDiscardIndex = userInputProvider.getNumberOfChosenCard();
 
         discardCardsIfPossible(player, firstCardToDiscardIndex, secondCardToDiscardIndex);
     }
 
     private void discardCardsIfPossible(Player player, int firstCardToDiscardIndex, int secondCardToDiscardIndex) {
-        if (player.areCardsMatching(firstCardToDiscardIndex, secondCardToDiscardIndex)) {
-            player.discardCardsFromHand(firstCardToDiscardIndex, secondCardToDiscardIndex);
-        }
-        else {
-            messagePrinter.printError(WRONG_CARDS_TO_DISCARD);
-            choseCardsToDiscard(player);
-        }
-    }
 
-    public int getValidCardIndex(Player player) {
-        int userInput = userInputProvider.getNumberOfChosenCard();
-        if (userInput > 0 && userInput <= player.getHand().size()) {
-            return (userInput - 1);
+        try {
```

Comment:
IMHO - it is easier to understand what is going on when you have this recurrence call in `choseCardsToDiscard` method. And you can remove `discardCardsIfPossible` method.
So you will have something like:
```java
    public void choseCardsToDiscard(Player player) {
        messagePrinter.printPlayer(player);

        messagePrinter.printMessage(ASK_FOR_CARDS_TO_DISCARD);
        int firstCardToDiscardIndex = userInputProvider.getNumberOfChosenCard();
        messagePrinter.printMessage(ASK_FOR_NEXT_CARD_TO_DISCARD);
        int secondCardToDiscardIndex = userInputProvider.getNumberOfChosenCard();

        try {
            if (player.areCardsMatching(firstCardToDiscardIndex, secondCardToDiscardIndex)) {
                player.discardCardsFromHand(firstCardToDiscardIndex, secondCardToDiscardIndex);
            } else {
                messagePrinter.printError(WRONG_CARDS_TO_DISCARD);
                choseCardsToDiscard(player);
            }
        } catch (IndexOutOfBoundsException exception) {
            messagePrinter.printError(exception.getMessage());
            choseCardsToDiscard(player);
        }
    }
```
so now when you are reading this method - you can easy spot the place where method is recursive (for `catch` case and `else`).

Also! Method name can be `discardCards` - you are not only choosing them but you are removing when indexes are correct and cards are matching.

Reply:
Much simpler